astropy/cosmology/_src/traits/darkenergy.py

```diff
@@ -0,0 +1,156 @@
+from abc import abstractmethod
+from math import exp, log
+from numbers import Number
+
+import numpy as np
+
+from astropy.cosmology._src.parameter import Parameter
+from astropy.cosmology._src.utils import aszarr, deprecated_keywords
+from astropy.utils.compat.optional_deps import HAS_SCIPY
+
+if HAS_SCIPY:
+    from scipy.integrate import quad
+else:
+
+    def quad(*args, **kwargs):
+        raise ModuleNotFoundError("No module named 'scipy.integrate'")
+
+
+class DarkEnergyComponent:
+    Ode0: Parameter = Parameter(
+        doc="Omega dark energy; dark energy density/critical density at z=0.",
+        fvalidate="float",
+    )
```

Comment:
I don't think we want to put the Parameter here. It should be a float. Then in FLRW it should be Parameter. This is so that subclasses don't need to have Ode0 as a Parameter.

Reply:
Thanks for the comment. Yeah makes perfect sense. I have pushed in the change. Replaced this with a comment to explain why its not here.